libp2p/errors.nim

```diff
@@ -5,6 +5,14 @@ import chronos
 import chronicles
 import macros
 
+# Uncomment only in dire need
+globalRaiseHook = proc (e: ref Exception): bool {.gcsafe, locks: 0.} =
```

Comment:
the usual way for these kinds of debugging helps is through a `-d` flag

Reply:
Sure, but I'm not even sure I want to push anything from this PR as I don't think it's something that needs to be in the repo but rather do it while developing. As I often do.

This PR is special cos I could not reproduce locally. Had to deploy and always wanted to share the logs.

Reply:
Do you guys actually think this helper should be in the code base behind something like `debug_exceptions` or so?

Reply:
Pushing is fine as long as it's behind a flag.

`debug_exceptions` is as good as any, or what I've been leaning towards is piggybacking from the chronicles log levels to some extent, for example I'm using trace level to enable most other debugging facilities info. Either way we don't and shouldn't for the most part log anything too verbose unless tracing.